gapic-generator-java/src/main/java/com/google/api/generator/gapic/model/Service.java

```diff
@@ -52,6 +52,12 @@ public boolean hasDescription() {
     return !Strings.isNullOrEmpty(description());
   }
 
+  public String hostServiceName() {
+    // Default Host is guaranteed to exist and be non-null and non-empty
```

Comment:
Ah, comment is misleading. I meant to write that the `hostServiceName is guaranteed` as the generator won't generate a library if it doesn't exist.

> I remember we discussed that the service name could come from the proto or service yaml, is that still true?

It is still true. Checks the proto's default_host value first, then the service yaml file, and fails otherwise